### PR TITLE
Autos

### PR DESCRIPTION
**Pull request by alicerobson**
addresses issue #14 
Note that I moved some of the sorting triggered by WITHSORT to make it easier to work without needing to copy.

### CODE REVIEW

File: papas/reconstruction/src/MergedClusterBuilder.cpp [hunk 1]
```diff
@@ -43,21 +43,17 @@ MergedClusterBuilder::MergedClusterBuilder(const Event& event,
...
-#if WITHSORT
-    ids.sort(std::greater<IdType>()); //sort in descending order
-#endif
-    auto id = *ids.begin();
+  GraphBuilder grBuilder(uniqueids, std::move(edges));
+  for (const auto& ids : grBuilder.subGraphs()) {
+    const auto& id = *ids.begin();
     double totalenergy = 0.;
-    for (const auto& c : ids) {
+    for (auto c : ids) {
```
Comment (cbernet) [inline papas/reconstruction/src/MergedClusterBuilder.cpp]: why changing that?

File: papas/display/src/Display.cpp [hunk 2]
```diff
@@ -14,7 +14,7 @@ Display::Display(std::list<ViewPane::Projection> views) {
     views = {ViewPane::Projection::xy, ViewPane::Projection::yz, ViewPane::Projection::xz};
   }
   /// Creates viewpanes
-  for (auto view : views) {
+  for (auto& view : views) {
```
Comment (alicerobson) [inline papas/display/src/Display.cpp]: In the end I decided not to handle the display stuff for this pull request as it is not really part of the library.

File: papas/graphtools/DirectedAcyclicGraph.h [hunk 3]
```diff
@@ -222,7 +222,7 @@ void BFSVisitor<N>::traverse(const Nodeset<N>& nodes, typename DAG::enumVisitTyp
   std::queue<int> nodeDepth;  // keeps track of the node depths so we can limit how deep we go if we wish
 
   // Mark the current node as visited and enqueue it
-  for (auto const& node : nodes) {
+  for (auto& node : nodes) {
```
Comment (alicerobson) [inline papas/graphtools/DirectedAcyclicGraph.h]: Yes, just got myself confused for a moment. Sorry. Have fixed so will arrive in a later pull request.

File: papas/reconstruction/src/MergedClusterBuilder.cpp (hunk 1 above)
Comment (alicerobson) [inline papas/reconstruction/src/MergedClusterBuilder.cpp]: c is just a long, so more consistent with elsewhere

**Pull request merged**